pyrene/src/components/TextArea/TextArea.tsx

```diff
@@ -89,10 +101,39 @@ const TextArea: React.FC<TextAreaProps> = ({
   onBlur,
   onChange,
   onFocus,
+  adaptToContent = false,
+  minRows = 1,
+  maxRows = 10,
 }: TextAreaProps) => {
 
+  const textAreaLineHeight = 12;
   const characterCount = maxLength - (value !== null ? value.length : 0);
   const characterLimitReached = characterCount < 0;
+  const textAreaRef = useRef<HTMLTextAreaElement>(null);
+  const [currentRows, setCurrentRows] = useState(minRows);
+  const [text, setText] = useState(value);
+
+  useEffect(() => {
+    if (adaptToContent && textAreaRef.current) {
+      // eslint-disable-next-line no-bitwise
+      const rowsFromHeight = ~~(textAreaRef.current.scrollHeight / textAreaLineHeight);
```

Comment:
I think `Math.floor()` is easier to read and `~~` starts to fail when dividing very large numbers.

Reply:
That makes sense, thanks :) Will fix it